var/spack/repos/builtin/packages/fftx/package.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import subprocess
+
+from spack import *
+
+
+class Fftx(CMakePackage):
+    """FFTX is the exascale follow-on to the FFTW open source discrete FFT
+    package for executing the Fast Fourier Transform as well as higher-level
+    operations composed of linear operations combined with DFT transforms."""
+
+    homepage = "https://spiral.net"
+    url      = "https://github.com/spiral-software/fftx/archive/0.9.0.tar.gz"
+    git      = "https://github.com/spiral-software/fftx.git"
+
+    maintainers = ['spiralgen']
+
+    version('develop', branch='develop')
+    version('main',  branch='main')
+    version('0.9.0', sha256='d4930e9b959fd56bb63543b359ca09a7ae2c56db3ffc28d7d3628d92fc79ce12')
+
+    variant('build_type', default='Release',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'),
+            description='The build type to build')
+
+    depends_on('spiral-software')
+    #  depends_on('spiral-package-fftx')
+    #  depends_on('spiral-package-simt')
+    #  depends_on('spiral-package-mpi')
+
+    @run_before('cmake')
+    def create_lib_source_code(self):
+        #  From directory examples/library run the build-lib-code.sh script
+        with working_dir(join_path(self.stage.source_path, 'examples/library')):
+            bld_script = join_path(self.stage.source_path,
```

Comment:
You can see how to use `which` to get an Executable for `bash` and use it at https://github.com/spack/spack/blob/6760175289e620fb4af6ff61d3a98116315af8e4/var/spack/repos/builtin/packages/flux-core/package.py#L129.

Reply:
Line 130 shows how to use `bash` to run the script.

Reply:
An example of running and providing command line options to your script can be found at:

https://github.com/spack/spack/blob/6760175289e620fb4af6ff61d3a98116315af8e4/var/spack/repos/builtin/packages/anaconda3/package.py#L59